pkg/controller/s3/bucket/replicationConfig.go

```diff
@@ -222,3 +216,13 @@ func (in *ReplicationConfigurationClient) Delete(ctx context.Context, bucket *v1
 	).Send(ctx)
 	return errors.Wrap(err, replicationDeleteFailed)
 }
+
+func sortReplicationRules(rules []awss3.ReplicationRule) {
+	if len(rules) != 0 {
```

Comment:
If `len(rules) == 0`, `for` loop wouldn't run at all so the function would be no-op. So, I think we can remove that `if` check here.